Tasks/Common/azurerest-common/azurerestutility.ts

```diff
@@ -1013,4 +993,53 @@ export async function createAppInsightsWebTest(endpoint, appInsightsResourceGrou
     });
 
     return deferred.promise;
+}
+
+/**
+ * Adds release annotation to an application insights resource
+ * 
+ * @param   endpoint                Service Principal Endpoint
+ * @param   appInsightsResourceId   ResourceId of the application insights resource
+ * @param   isDeploymentSuccess     Add annotation for success or failed deployment
+ */
+export async function addReleaseAnnotation(endpoint, appInsightsResourceId: string, isDeploymentSuccess: string): Promise<any> {
+    let deferred: Q.Deferred<any> = Q.defer<any>();
+
+    let accessToken = await getAuthorizationToken(endpoint);
+    let restOptions: IRequestOptions = {
+        additionalHeaders: {
+            "Authorization": `Bearer ${accessToken}`
+        }
+    };
+
+    let restUrl = `${endpoint.url}${appInsightsResourceId}/Annotations?api-version=2015-05-01`;
```

Comment:
should addition of annotations be excluded for Azure stack? @Ajay-MS , can you confirm?

Reply:
@asranja - Do we know if this works for other Cloud types apart from Azure Cloud?

Reply:
No this does not work for any other cloud type or azure stack currently.